### PR TITLE
RF: Move tracking branch update functionality into GitRepo.update_submodule()

### PR DESCRIPTION
**Pull request by mih**
This implements a long-standing TODO. It was originally part of https://github.com/datalad/datalad/pull/3828 but I moved it out, because it may need more work.

### CODE REVIEW

File: datalad/support/gitrepo.py [hunk 1]
```diff
@@ -2760,6 +2760,43 @@ def update_submodule(self, path, mode='checkout', init=False):
...
         self._git_custom_command(path, cmd)
+
+        if not init:
+            return
+
+        # track branch originally cloned
+        branch = self.get_active_branch()
+        branch_hexsha = self.get_hexsha(branch)
+        updated_branch = self.get_active_branch()
+        if branch and not updated_branch:
```
Comment (bpoldrack) [inline datalad/support/gitrepo.py]: This is not plausible to me. In the original code block there was a submodule-update in between the two calls to `get_active_branch`. But here - How would `branch` and `updated_branch` ever differ from each other?

Comment (mih) [reply datalad/support/gitrepo.py]: Arrg, thx for spotting that. Pretty much the entire change was non-sensicle -- pushed an update now.

**Comment by mih** [comment]
This is ready, I think.

**Comment by bpoldrack** [comment]
Looks right to me.

**Comment by mih** [comment]
Thanks for the test @kyleam!

**Pull request merged**